integration-tests/src/test/groovy/com/okta/sdk/tests/it/AuthorizationServerIT.groovy

```diff
@@ -43,10 +55,463 @@ class AuthorizationServerIT extends ITSupport {
+                .setDescription("Test Authorization Server")
+                .setAudiences(["api://default"]))
+        registerForCleanup(createdAuthorizationServer)
+
+        assertThat(createdAuthorizationServer, notNullValue())
+        assertPresent(client.listAuthorizationServers(), createdAuthorizationServer)
+    }
+
+    @Test
+    void getAuthorizationServerTest() {
+        String name = "java-sdk-authorization-server-" + RandomStringUtils.randomAlphanumeric(15)
+
+        AuthorizationServer createdAuthorizationServer = client.createAuthorizationServer(
+            client.instantiate(AuthorizationServer)
+                .setName(name)
+                .setDescription("Test Authorization Server")
+                .setAudiences(["api://default"]))
+        registerForCleanup(createdAuthorizationServer)
+
+        assertThat(createdAuthorizationServer, notNullValue())
+
+        AuthorizationServer retrievedAuthorizationServer = client.getAuthorizationServer(createdAuthorizationServer.getId())
+        assertThat(retrievedAuthorizationServer.getId(), equalTo(createdAuthorizationServer.getId()))
+    }
+
+    @Test
+    void updateAuthorizationServerTest() {
+        String name = "java-sdk-authorization-server-" + RandomStringUtils.randomAlphanumeric(15)
+
+        AuthorizationServer createdAuthorizationServer = client.createAuthorizationServer(
+            client.instantiate(AuthorizationServer)
+                .setName(name)
+                .setDescription("Test Authorization Server")
+                .setAudiences(["api://default"]))
+        registerForCleanup(createdAuthorizationServer)
+
+        assertThat(createdAuthorizationServer, notNullValue())
+
+        createdAuthorizationServer.setDescription("Updated Test Authorization Server")
+
+        AuthorizationServer updatedAuthorizationServer = createdAuthorizationServer.update()
+        assertThat(updatedAuthorizationServer.getId(), equalTo(createdAuthorizationServer.getId()))
+        assertThat(updatedAuthorizationServer.getDescription(), equalTo("Updated Test Authorization Server"))
+    }
+
+    @Test
+    void deleteAuthorizationServerTest() {
+        String name = "java-sdk-authorization-server-" + RandomStringUtils.randomAlphanumeric(15)
+
+        AuthorizationServer createdAuthorizationServer = client.createAuthorizationServer(
+            client.instantiate(AuthorizationServer)
+                .setName(name)
+                .setDescription("Test Authorization Server")
+                .setAudiences(["api://default"]))
+        registerForCleanup(createdAuthorizationServer)
+
+        assertThat(createdAuthorizationServer, notNullValue())
+        assertThat(client.getAuthorizationServer(createdAuthorizationServer.getId()), notNullValue())
+
+        createdAuthorizationServer.deactivate()
+        createdAuthorizationServer.delete()
+
+        sleep(2000)
+
+        try {
+            client.getAuthorizationServer(createdAuthorizationServer.getId())
```

Comment:
you can also use something like:

```groovy
ResourceException e = expectException ResourceException, { client.getAuthorizationServer(createdAuthorizationServer.getId()) }
assertThat(e.status, equalTo(404))
```

Reply:
will do

Reply:
Did you want to assert on the status code?

Reply:
Updated with an alternate way of verifying. List the resources and check for absence of the resource in question.